examples/build/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	v1 "github.com/openshift/api/build/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	buildv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+	"path/filepath"
+	"flag"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"fmt"
+)
+
+func main() {
+	var kubeconfig *string
+	if home := homeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
```

Comment:
This doesn't provide a way to register the flags?

Reply:
I copied the code from k8s go-client example:
https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go
Also searched a bit on the code there, did not find any.